src/Components/CookieConsent.js

```diff
@@ -1,47 +1,48 @@
 import * as React from "react";
 import * as Scrivito from "scrivito";
-import ReactCookieConsent, { Cookies } from "react-cookie-consent";
+import {
+  cookieConsentGiven,
+  acceptCookieConsent,
+  cookieConsentUrl,
+  declineCookieConsent,
+  getCookieConsent,
+} from "../utils/cookieConsentManagement";
 import cookieConsentIcon from "../assets/images/cookie_consent_icon.svg";
-import { resolveCookieConsent } from "../utils/cookieConsentGiven";
 
-class CookieConsent extends React.Component {
-  componentDidMount() {
-    if (Cookies.get("CookieConsent") === "true") {
-      resolveCookieConsent();
-    }
-  }
+function CookieConsent() {
+  const consentUrl = cookieConsentUrl();
+  const [visible, setVisible] = React.useState(false);
 
-  render() {
-    const root = Scrivito.Obj.root();
+  React.useEffect(() => {
+    setVisible(getCookieConsent() === "undecided");
 
-    if (!root) {
-      return null;
-    }
+    cookieConsentGiven().then(() => {
+      setVisible(false);
```

Comment:
* [ ] FYI we're rerendering `null` for no reason in most cases (all except `undecided`). I don't know if this has any measurable effect.

Reply:
I don't understand the question. What do you suggest or what is your concern?

Reply:
If we would like minimise the number of re-renders, we should call `setVisible` only if the arg differs from the current `visible` value.